### PR TITLE
MediaElement: Fix duplicate audio icons

### PR DESCRIPTION
**Pull request by crhallberg**
Example Manifest: https://digital.library.villanova.edu/Item/vudl:1006929/Manifest.

Upstream PR opened: https://github.com/mediaelement/mediaelement-plugins/pull/266.

---

If you load an audio manifest with multiple sources into the UV, [the source selector is also a play button icon](https://github.com/user-attachments/assets/62431168-789d-45d9-b0db-3d81c4c90aa6).

## Investigation

It looks like the button icons have been replaced locally with a modified version of the mediaelementplayer.css. This seems to be a fix for the buttons having no icons without this addition. The icons are missing because of [some normalization styles for unstyled buttons on Android](f774e64eb77ec95121572559a76c67f7ea3b10a3). Even sneakier, this normalization also [exists as a reset in TailwindCSS](https://github.com/tailwindlabs/tailwindcss/blob/next/packages/tailwindcss/preflight.css#L193), which was [used to build the example page](https://github.com/UniversalViewer/universalviewer/blob/main/src/index.html#L9-L11). So...

## Repairs

1. Remove the normalizations.
1. Use the prebuilt CSS for mediaelementplayer.
1. Refactor the example page styles to remove TailwindCSS.

## Enhancements

1. Error reduction in source-chooser.
2. Keyboard accessibility improvements for source-chooser.

## Concerns

This PR contains a fork of the source-chooser mediaelement plugin. I am going to contribute this upstream once it's been tested and verified, but [this part of the project looks long in the tooth...](https://github.com/mediaelement/mediaelement-plugins/tree/master/dist/source-chooser).

### CODE REVIEW

**Comment by demiankatz** [comment]
Thanks, @crhallberg, this certainly looks like an improvement to me.

If anyone else wants to test, you can use this manifest: https://digital.library.villanova.edu/Item/vudl:384795/Manifest

A couple of notes/comments:

1.) I feel like the keyboard behavior still feels a little weird (for example, when I popped open the source selector, I expected to be able to use the arrow keys but instead had to use tab)... but it's still forward progress to move from "completely broken" to "a little weird." Is this related to the part you hope to contribute upstream?

2.) And speaking of the upstream contribution, is the plan to test here, then contribute upstream, then merge this after the upstream contributions become available? Or do we want to merge with the local override file in the interim? I suppose the answer to that may be dependent on how things play out upstream. My preference would be to avoid having the local override file in the project if we can, but I recognize that it may be our only option if upstream adoption is too slow. Whatever happens, I just want to be sure we don't forget about this and leave the work half-completed indefinitely!

**Comment by LanieOkorodudu** [comment]
Thanks @crhallberg, The source selector icon looks much better now, and it's definitely easier for users to recognize. When I tested it, I was able to use the ↑ and ↓ arrow keys to select a source. However, I agree with @demiankatz  about needing to press the Tab key first before using the arrow keys.

One concern is that users who expect to immediately use the ↑ and ↓ arrow keys might not realize they need to press Tab first. Would it be helpful to display a brief pop-up message like "Press Tab, then use ↑ and ↓ arrow keys" when navigating to the source chooser? This would assist sighted users, and as UV's accessibility improves, a screen reader could convey this as well. Just a suggestion!

Overall, this is a fantastic improvement, making keyboard navigation much more user-friendly

**Comment by demiankatz** [comment]
@LanieOkorodudu, I found that the tab/arrow behavior became much more intuitive after @crhallberg's most recent commit (https://github.com/UniversalViewer/universalviewer/pull/1130/commits/83318ce0ec3605189feb124f427c9e4d51138757). It's possible there are still ways of making it more accessible, but to my non-expert perception, it seems good enough for now. I'd be happy to see what happens if this version is submitted upstream to MediaElement, though of course I'm happy to defer to others with more expertise if further adjustment is still needed.

**Comment by LanieOkorodudu** [comment]
> @LanieOkorodudu, I found that the tab/arrow behavior became much more intuitive after @crhallberg's most recent commit ([83318ce](https://github.com/UniversalViewer/universalviewer/commit/83318ce0ec3605189feb124f427c9e4d51138757)). It's possible there are still ways of making it more accessible, but to my non-expert perception, it seems good enough for now. I'd be happy to see what happens if this version is submitted upstream to MediaElement, though of course I'm happy to defer to others with more expertise if further adjustment is still needed.

@demiankatz, The fix looks good to me too! The suggestion was just an idea—the main priority is that we're able to use the keyboard for navigation effectively. I'm happy with this fix and it's good to merge.

**Comment by crhallberg** [comment]
Upstream PR opened: https://github.com/mediaelement/mediaelement-plugins/pull/266.

**Comment by LanieOkorodudu** [comment]
@crhallberg, Yes, the expectation is that when the settings icon is focused, pressing Enter (or Space) would open the menu and highlight the first option, allowing immediate use of the up and down arrow keys to navigate through options.

Currently, I have to press Tab first before the up and down arrows work to select a source, and Enter/Space isn’t triggering the menu selection as expected.
Would it be possible to update the behaviour so that pressing Enter on the focused settings icon brings focus directly to the source settings, with the first option highlighted? This would allow smooth navigation with the arrow keys from there.

**Comment by crhallberg** [comment]
@LanieOkorodudu this is the intended control flow. What browser are you using for testing?

**Comment by LanieOkorodudu** [comment]
@crhallberg 

https://github.com/user-attachments/assets/26c1afef-0545-4af2-b1f2-c80420eeb837

Pressing the Tab until the focus highlight in the source chooser setting, Then Press Enter to bring up a list or menu displaying the different source options to choose from. After that I press down and arrow keys, this should cycle through the different source option through the different source options within the selector.

If you see in the video, when I press the down and arrow keys changes the volume instead of moving through source options.

When I tested this last week the up and down arrow key works fine. I am using chrome and Firefox. But since the fix was to change the source chooser icon 

> @crhallberg, Yes, the expectation is that when the settings icon is focused, pressing Enter (or Space) would open the menu and highlight the first option, allowing immediate use of the up and down arrow keys to navigate through options.
> 
> Currently, I have to press Tab first before the up and down arrows work to select a source, and Enter/Space isn’t triggering the menu selection as expected. Would it be possible to update the behaviour so that pressing Enter on the focused settings icon brings focus directly to the source settings, with the first option highlighted? This would allow smooth navigation with the arrow keys from there.

 @crhallberg I tested this using  edge. I did some testing today in Firefox and chrome will upload the videos here.

**Comment by LanieOkorodudu** [comment]
@crhallberg 

https://github.com/user-attachments/assets/ac6d9ea4-641a-4e16-9f16-19354dd46c7f

I press Tab until the focus highlights the source chooser setting, then press Enter to bring up a menu displaying the available source options. Next, I use the Down and Up Arrow keys, which should allow me to cycle through the different source options within the selector.

However, as shown in the video, pressing the Down and Up Arrow keys changes the volume instead of moving through the source options.

Additionally, when I press Enter today, it correctly triggers the source selection, which didn’t happen last week. I tested this behaviour in both Chrome and Firefox.

Since the recent fix was to change the source chooser icon, this specific issue with the arrow keys may be unrelated and could require a separate fix..

**Comment by demiankatz** [comment]
Thanks, @LanieOkorodudu! I think we're essentially done here, but since I'm still hoping we can get the upstream fix merged to mediaelement, I'm not going to merge this yet. If we don't get any traction there by next week's Community Call, we can discuss there whether it's better to merge this fix with the local override, or continue to wait.

File: src/content-handlers/iiif/modules/uv-mediaelementcenterpanel-module/js/source-chooser-fixed.js
Comment (demiankatz) [inline src/content-handlers/iiif/modules/uv-mediaelementcenterpanel-module/js/source-chooser-fixed.js:12]: Maybe this comment should be replaced with a link to the upstream pull request, if we're contemplating merging this as-is due to lack of upstream activity. That way, if in future we try to get rid of this file, we understand why it's here and how to know when it's no longer needed.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Unfortunately there's been no motion in the upstream PR related to this one. I think maybe it's time to give up and merge our custom version of the code in the interest of fixing the issue. If we do that, though, I have one minor suggestion first....

**Pull request merged**
